### ISSUE TITLE
Adding specific start/end dates for MP party membership

### ISSUE THREAD

**Issue by fredrik1984** [welfare-state-analytics/riksdagen-corpus#428]
Currently, start/end of MP party membership are based on years (e.g. 1910–1923). However, some MPs could be a member of a party for less than a year and then join another (like this MP: https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABfQug). Then it will be more messy (e.g. 1907–1907).

In the bio books, only years are stated for party start/end. For the Swerik project, it is for now a low priority to add specific MP party start/end dates. One reason for this is that it could be very difficult to assess when an MP started/ended in a party. 

However, @ljo had some ideas of how this could be approached in a structured way.

**Comment by salgo60** [comment]
@fredrik1984 what source Will you have for this data?

I guess you still will have problems as I feel som "party" affiliations are made up and are more constructions in 1985 and was not used in early 1900 as I show in my issue [139 about vilde](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/139) --> the challenge is to state that different sources state different facts see #421 how Wikidata support that....

**Comment by MansMeg** [comment]
I agree. This isn't easy how to solve. We should focus on the party affiliation when the persons are members of parliament. Hence, the records might be a good source for this. 

Otherwise, we will need to look into other sources. I think the librarians at the Parliament Library probably have some information on good sources for this. Also, the further back we go, the more loose the idea of a party will be. Before 1730, I don't think there existed parties in the parliament. Since we want only to have parties defined by a source it might become harder and harder to define when people enter and leave parties.

Although, I think @ljo might have some ideas how to do this in a good way.

**Comment by salgo60** [comment]
for [member of the Riksdag of the Estates](https://www.wikidata.org/wiki/Q82697153) wikidata use [P1268 represents](https://www.wikidata.org/wiki/Property:P1268) see [Q343674#P39](https://www.wikidata.org/wiki/Q343674#P39)

<img width="1169" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/ec90146c-812b-4c8b-9738-b49d6fcdbe0e">

[Timeline where we have that P1268 in WD](https://js.histropedia.com/apps/query-timeline/index.html?q=SELECT%20distinct%20?person%20?personLabel%20%20(sample(?bild)%20AS%20?bild)%20%0A?birth%20?death%20?representsLabel%20WHERE%20%7B%0A%0A%20%20VALUES%20?member%20%7B%0A%20%20%20%20wd:Q33071890%20%0A%20%20%20%20wd:Q81531912%20%0A%20%20%20%20wd:Q82697153%20%0A%20%20%20%20wd:Q10655178%20%0A%20%20%7D%0A%20%20%7B?person%20p:P39%20?pMember.%0A%20%20%20?pMember%20ps:P39%20?member.%0A%20%20%20?pMember%20pq:P1268%20?represents.%20%23represents%0A%20%20%7D%0A%20%20OPTIONAL%7B?person%20wdt:P569%20?birth%7D%0A%20%20OPTIONAL%7B?person%20wdt:P570%20?death%7D%0A%20%20%0A%0A%20%20OPTIONAL%20%7B?person%20wdt:P18%20?bild%7D.%0A%20%0A%20%20SERVICE%20wikibase:label%20%7B%20bd:serviceParam%20wikibase:language%20%22sv,en%22.%20%7D%0A%7D%20GROUP%20BY%20%20?person%20?personLabel%20%20?death%20?birth%20?representsLabel%0AORDER%20BY%20?representsLabel&md=true&g=article&l=person&t=personLabel&s=birth&e=death&i=bild&d=0&c=representsLabel&f=representsLabel&v=t) / [SPARQL](https://w.wiki/8MSE)

<img width="1426" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/883f9788-b2c9-4ee7-aaf0-bfcf8c82df3e">

* [Timeline showing those Swedish MPs with parties and represents](https://js.histropedia.com/apps/query-timeline/index.html?q=%23title:%20Party%20for%20people%20with%20represents%20P1268%0ASELECT%20distinct%20?person%20?personLabel%20%20(sample(?bild)%20AS%20?bild)%20%0A?birth%20?death%20?representsLabel%20%20?partyLabel%20WHERE%20%7B%0A%0A%20%20VALUES%20?member%20%7B%0A%20%20%20%20wd:Q33071890%20%0A%20%20%20%20wd:Q81531912%20%0A%20%20%20%20wd:Q82697153%20%0A%20%20%20%20wd:Q10655178%20%0A%20%20%7D%0A%20%20%7B?person%20p:P39%20?pMember.%0A%20%20%20?pMember%20ps:P39%20?member.%0A%20%20%20?pMember%20pq:P1268%20?represents.%20%23represents%0A%20%20%7D%0A%20%20OPTIONAL%7B?person%20wdt:P569%20?birth%7D%0A%20%20OPTIONAL%7B?person%20wdt:P570%20?death%7D%0A%20%20OPTIONAL%7B?person%20wdt:P18%20?bild%7D%0A%20%20%20%7B?person%20wdt:P102%20?party%7D.%0A%20%0A%20%20SERVICE%20wikibase:label%20%7B%20bd:serviceParam%20wikibase:language%20%22sv,en%22.%20%7D%0A%7D%20GROUP%20BY%20%20?person%20?personLabel%20%20?death%20?birth%20?representsLabel%20?partyLabel%0AORDER%20BY%20?partyLabel&d=0&md=true&g=article&l=person&t=personLabel&s=birth&e=death&i=bild&c=partyLabel&f=partyLabel&v=t)

<img width="1356" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/41a48b5a-257f-4418-b543-255052b889c9">

### Wikidata:WikiProject_British_Politicians

I guess best maintained in WD is [Wikidata:WikiProject_British_Politicians](https://www.wikidata.org/wiki/Wikidata:WikiProject_British_Politicians)
* [sample queries](https://www.wikidata.org/wiki/Wikidata:WikiProject_British_Politicians/Sample_Queries)
* [data model](https://www.wikidata.org/wiki/Wikidata:WikiProject_British_Politicians/Data_model)
* [coverage](https://www.wikidata.org/wiki/Wikidata:WikiProject_British_Politicians/Coverage)

<img width="1426" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/fb9ceada-df9a-4450-9601-540179744ad5">
